view/frontend/web/js/button-config-platform.js

```diff
@@ -5,290 +5,318 @@
-                            if (typeof this.config.shippingOptions === "undefined" && quote.isVirtual()) {
-                                this.config.customTotal = this.round(quote.getTotals()._latestValue.base_grand_total);
-                                cb();
+                                bread_sdk.on('INSTALLMENT:APPLICATION_DECISIONED', self.config.onApproval);
+                                bread_sdk.on('INSTALLMENT:APPLICATION_CHECKOUT', self.config.onCheckout);
+                                bread_sdk.init();
+                                this.wasSetup = true;
                             } else {
-                                /* ocs save selected shipping method */
-                                var shippingOptionUrl = window.checkoutConfig.payment.breadcheckout.shippingOptionUrl;
-                                $.ajax(
-                                        {
-                                            url: shippingOptionUrl,
-                                            type: 'post',
-                                            context: this
-                                        }
-                                ).done(
-                                        function (data) {
-                                            self.config.shippingOptions = [data];
-                                            self.config.customTotal = self.round(quote.getTotals()._latestValue.base_grand_total);
-                                            cb();
-                                        }
-                                ).fail(
-                                        function (error) {
-                                            var errorInfo = {
-                                                bread_config: self.config
-                                            };
-                                            document.logBreadIssue(
-                                                    'error', errorInfo,
-                                                    'Error code returned when calling ' + shippingOptionUrl + ', with status: ' + error.statusText
-                                                    );
-                                        }
-                                );
+                                bread_sdk.registerPlacements([placementObject]);
+                                bread_sdk.openExperienceForPlacement([placementObject]);
                             }
+                            fullScreenLoader.stopLoader();
+                        }
+                    });
+                }
+            },
+            createBopisPayload: function(fulfillmentType, hasPickupInformation = true) {
+                var self = this;
+                if (fulfillmentType === 'PICKUP' && hasPickupInformation) {
+                    return {
+                        pickupInformation: {
+                            name: {
+                                givenName: self.config.billingContact.firstName,
```

Comment:
Is billing given name a required field in Magento?  We had an issue in shopify where the first name on the billing address wasn't required, leading to errors finishing the checkout.

Reply:
@dschoder-bread Good call. It is a required field in vanilla version of Magento. However, there are third party extensions like one-step-checkout that change the layout and behavior of some of those fields. How did you fix it in shopify? Did you remove `name` from `pickupInformation` if first name is not set?

Reply:
Yeah, if you remove `name` and `phone` then the modal will use what you enter in the bread modal's billing info page.